core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java

```diff
@@ -197,4 +199,15 @@ public void testEnforceLimits() {
         assertTrue(BootstrapCheck.enforceLimits(settings));
     }
 
+    public void testMinMasterNodes() {
+        BootstrapCheck.Check check = new BootstrapCheck.MinMasterNodesCheck();
+        assertTrue(check.check(Settings.EMPTY));
+        assertFalse(check.check(
+            Settings.builder().put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), 1).build()
+        ));
+        expectThrows(RuntimeException.class, () ->
+            BootstrapCheck.check(
```

Comment:
Yeah, let's the keep the tests just focused on whether or not `MinMasterNodeCheck` does the right thing based on whether or not `discovery.zen.minimum_master_nodes` is set and we can think about broader tests for the default checks from `BootstrapCheck` itself in a separate pull request.